src/OnChain.hs

```diff
@@ -66,6 +70,7 @@ updateRegistration ScriptParams{..} sig ctx
         opOk d1 d2 = (pEnOpCs d1) == (pEnOpCs d2) && (enUsedNftTn d1) == (enUsedNftTn d2)
 
         -- Check if the new datum was signed by the current CceAddress, the signature must be provided by the redeemer
+        -- TODO Not sure, but is this redundant with the `checkDatumSig nDat` check?
```

Comment:
the "checkDatumSig" takes all inputs from the datum passed to the function; here we take the newDatum for verification, but the PubKey is taken from the current (old) Datum, so we can be sure the signer of the new datum is the same who signed the old datum.